src/makeInfinityAdder.js

```diff
@@ -6,7 +6,32 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let count = 0;
+
+  function adder(x) {
+    if (x !== undefined) {
+      sum += x;
+
+      return add;
+    }
+    count++;
+    sum = count > 1 ? 0 : sum;
+
+    return sum;
+  };
```

Comment:
We can return adder function inside itself, no need to create a separate function add